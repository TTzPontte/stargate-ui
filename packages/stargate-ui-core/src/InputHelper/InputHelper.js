import React from 'react';
import PropTypes from 'prop-types';

import Typography from '../Typography';

/**
 * @todo needs to add support for custom element.
 */
const InputHelper = (props) => {
  const {
    children,
    color,
    ...factoryProps
  } = props;

  return (
    <Typography
      element="small"
      variant="small"
      display="block"
      marginBottom={1}
      color={color}
      children={children}
      {...factoryProps}
    />
  );
};

InputHelper.propTypes = {
  children: PropTypes.string.isRequired,
  color: PropTypes.oneOf([
    'default',
    'success',
    'warning',
    'error',
  ]),
};

export default InputHelper;
