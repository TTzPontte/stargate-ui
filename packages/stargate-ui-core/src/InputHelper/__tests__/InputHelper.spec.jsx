import React from 'react';

import InputHelper from '../InputHelper';

describe('InputHelper', () => {
  it('should render properly', () => {
    const { container } = render(
      <InputHelper>
        Lorem ipsum dolor siet amiet.
      </InputHelper>
    );

    expect(container).toMatchSnapshot();
  });
});
