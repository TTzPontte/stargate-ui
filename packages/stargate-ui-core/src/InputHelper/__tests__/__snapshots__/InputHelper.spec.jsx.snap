// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelper should render properly 1`] = `
<div>
  <small
    class="stargate-factory-0-2-4 stargate-factory-d0-0-2-5 stargate-typography-0-2-2 stargate-typography-d0-0-2-3"
  >
    Lorem ipsum dolor siet amiet.
  </small>
</div>
`;
